android/src/main/java/com/amazonlogin/RNAmazonLoginModule.java

```diff
@@ -0,0 +1,172 @@
+      @Override
+      public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(activity, requestCode, resultCode, data);
+      }
+    };
+    reactContext.addLifecycleEventListener(this);
+    reactContext.addActivityEventListener(mActivityEventListener);
+  }
+
+  @Override
+  public String getName() {
+    return "RNAmazonLogin";
+  }
+
+  @ReactMethod
+  public void login(final Promise promise) {
+    mAuthPromise = promise;
+    AuthorizationManager.authorize(
+                        new AuthorizeRequest.Builder(requestContext)
+                                .addScopes(ProfileScope.profile(), ProfileScope.postalCode())
+                                .build()
+                );
+  }
+
+  @ReactMethod
+  public void isAuthorized(final Promise promise) {
+    Scope[] scopes = {ProfileScope.profile(), ProfileScope.postalCode()};
+    AuthorizationManager.getToken(getCurrentActivity(), scopes, new Listener<AuthorizeResult, AuthError>() {
+        @Override
+        public void onSuccess(AuthorizeResult authorizeResult) {
+            if (authorizeResult.getAccessToken() != null) {
+                /* The user is signed in */
+                final WritableMap userMap = Arguments.createMap();
+                User user = authorizeResult.getUser();
+                userMap.putString("id", user.getUserId());
+                userMap.putString("name", user.getUserName());
+                userMap.putString("email", user.getUserEmail());
+                userMap.putString("postalCode", user.getUserPostalCode());
+
+                final WritableMap result = Arguments.createMap();
+                result.putString("token", authorizeResult.getAccessToken());
+                result.putMap("user", userMap);
+                promise.resolve(result);
+            } else {
+                /* The user is not signed in */
+                promise.resolve(null);
+            }
+        }
+
+        @Override
+        public void onError(AuthError authError) {
+            /* The user is not signed in */
+            promise.reject(TAG, authError.getMessage());
+        }
+    });
+  }
+
+  @ReactMethod
+  public void logout(final Promise promise) {
+    AuthorizationManager.signOut(getReactApplicationContext().getApplicationContext(), new Listener<Void, AuthError>() {
+      @Override
+      public void onSuccess(Void response) {
+        promise.resolve(response);
+      }
+
+      @Override
+      public void onError(AuthError authError) {
+          promise.reject(TAG, authError.getMessage());
+      }
+  });
+  }
+
+  @Override
+  public void initialize() {
+    requestContext = RequestContext.create(getCurrentActivity());
+    requestContext.registerListener(new AuthorizeListener() {
+      /* Authorization was completed successfully. */
+      @Override
+      public void onSuccess(AuthorizeResult authorizeResult) {
+        final WritableMap userMap = Arguments.createMap();
```

Comment:
Next 10 lines are copy-paste. Extract to separate method for example AuthorizeResultToMap or smth like that.